packages/frontend/src/components/card_list.tsx

```diff
@@ -0,0 +1,77 @@
+import type { QueryOptions } from "@esp-group-one/types";
+import type { ReactNode } from "react";
+import { useEffect, useState } from "react";
+
+interface CardListProps<T extends ReactNode> {
+  pageSize: number;
+  query: unknown;
```

Comment:
This is going to be changing in https://github.com/ESP-Number-One/MatchyoSports/pull/35... might be best to talk this over in person